autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -0,0 +1,533 @@
+                 exclude: Optional[Dict[str, Any]] = None,
+                 disable_file_output: bool = False,
+                 init_params: Optional[Dict[str, Any]] = None,
+                 budget: Optional[float] = None,
+                 budget_type: Optional[str] = None,
+                 logger_port: Optional[int] = None) -> None:
+
+        self.starttime = time.time()
+
+        self.configuration = configuration
+        self.backend = backend
+        self.queue = queue
+
+        self.datamanager: BaseDataset = self.backend.load_datamanager()
+        self.include = include
+        self.exclude = exclude
+
+        self.X_train, self.y_train = self.datamanager.train_tensors
+
+        if self.datamanager.val_tensors is not None:
+            self.X_valid, self.y_valid = self.datamanager.val_tensors
+        else:
+            self.X_valid, self.y_valid = None, None
+
+        if self.datamanager.test_tensors is not None:
+            self.X_test, self.y_test = self.datamanager.test_tensors
+        else:
+            self.X_test, self.y_test = None, None
+
+        self.metric = metric
+        self.task_type = STRING_TO_TASK_TYPES[self.datamanager.task_type]
+        self.output_type = STRING_TO_OUTPUT_TYPES[self.datamanager.output_type]
+        self.issparse = self.datamanager.issparse
+
+        self.seed = seed
+
+        self.output_y_hat_optimization = output_y_hat_optimization
+
+        if isinstance(disable_file_output, (bool, list)):
+            self.disable_file_output = disable_file_output
+        else:
+            raise ValueError('disable_file_output should be either a bool or a list')
+
+        self.pipeline_class: Optional[BaseEstimator] = None
+        info = {'task_type': self.datamanager.task_type,
+                'output_type': self.datamanager.output_type,
+                'issparse': self.issparse}
+        if self.task_type in REGRESSION_TASKS:
+            if not isinstance(self.configuration, Configuration):
+                self.pipeline_class = DummyRegressionPipeline
+            else:
+                if self.task_type in TABULAR_TASKS:
+                    self.pipeline_class = TabularRegressionPipeline
+                else:
+                    raise ValueError('task {} not available'.format(self.task_type))
+            self.predict_function = self._predict_regression
+        else:
+            if not isinstance(self.configuration, Configuration):
+                self.pipeline_class = DummyClassificationPipeline
+            else:
+                if self.task_type in TABULAR_TASKS:
+                    self.pipeline_class = TabularClassificationPipeline
+                elif self.task_type in IMAGE_TASKS:
+                    self.pipeline_class = ImageClassificationPipeline
+                else:
+                    raise ValueError('task {} not available'.format(self.task_type))
+            self.predict_function = self._predict_regression
```

Comment:
Should this be `self.predict_function = self._predict_classification`?

Reply:
yeah we need a predict_proba for classification, which we dont have so I created an issue for it and in the time being just using the preidct_regression as it just calls the model predict